artifactory/services/security.go

```diff
@@ -23,10 +27,42 @@ func NewSecurityService(client *rthttpclient.ArtifactoryHttpClient) *SecuritySer
 	return &SecurityService{client: client}
 }
 
+// GetAPIKeyPath return the apiKeyPath
+func GetAPIKeyPath() string {
+	return apiKeyPath
+}
+
 func (ss *SecurityService) getArtifactoryDetails() auth.ArtifactoryDetails {
 	return ss.ArtDetails
 }
 
+// RegenerateAPIKey regenerates the API Key in Artifactory
+func (ss *SecurityService) RegenerateAPIKey() (string, error) {
+	httpClientDetails := ss.ArtDetails.CreateHttpClientDetails()
+
+	reqURL, err := utils.BuildArtifactoryUrl(ss.ArtDetails.GetUrl(), apiKeyPath, nil)
+	if err != nil {
+		return "", err
+	}
+
+	resp, body, err := ss.client.SendPut(reqURL, nil, &httpClientDetails)
+	if err != nil {
+		return "", err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return "", errors.New("api key reneration failed with status: " + resp.Status + "\n" + clientutils.IndentJson(body))
+	}
+
+	var data map[string]interface{} = make(map[string]interface{})
+	if err := json.Unmarshal(body, &data); err != nil {
+		return "", fmt.Errorf("unable to decode json from response: %w", err)
```

Comment:
unable --> Unable
Let's also add the actual response received. Could come in handy when such an error occurs.